From 79e8330735b8de06554c16dacf0b18a979401c90 Mon Sep 17 00:00:00 2001
From: Dexter Rivera <dexter_180@live.com>
Date: Wed, 24 Jun 2020 08:02:26 -0700
Subject: [PATCH 1/1] patch makefile

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 4d1dde1..7c9fc4a 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,7 +1,7 @@
 SHELL = /bin/sh
 TOPDIR = $(realpath ..)

-CPPFLAGS = -Iinclude -I/usr/include/json-c
+CPPFLAGS = -Iinclude -I$(JSON_INCLUDE_PATH)
 FLAGS = -fPIC -Wall -g
 CFLAGS = $(FLAGS) -Wstrict-prototypes
 CXXFLAGS = $(FLAGS)
--
2.27.0.111.gc72c7da667-goog

