From 1f3ff12272368b2b91daa4284de1607c88a69d75 Mon Sep 17 00:00:00 2001
From: Wei Xu <weixugc@google.com>
Date: Tue, 21 Jan 2020 17:35:43 -0800
Subject: [PATCH] Remove sshd.service dependencies from cloud-init.service.

---
 systemd/cloud-init.service.tmpl | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/systemd/cloud-init.service.tmpl b/systemd/cloud-init.service.tmpl
index 9ad3574c..6a8592ed 100644
--- a/systemd/cloud-init.service.tmpl
+++ b/systemd/cloud-init.service.tmpl
@@ -3,8 +3,6 @@
 Description=Initial cloud-init job (metadata service crawler)
 DefaultDependencies=no
 Wants=cloud-init-local.service
-Wants=sshd-keygen.service
-Wants=sshd.service
 After=cloud-init-local.service
 After=systemd-networkd-wait-online.service
 {% if variant in ["ubuntu", "unknown", "debian"] %}
@@ -21,8 +19,6 @@ After=wicked.service
 After=dbus.service
 {% endif %}
 Before=network-online.target
-Before=sshd-keygen.service
-Before=sshd.service
 {% if variant in ["ubuntu", "unknown", "debian"] %}
 Before=sysinit.target
 Conflicts=shutdown.target
-- 
2.25.1.481.gfbce0eb801-goog

